pkg/token/token.go

```diff
@@ -570,7 +578,11 @@ func (v tokenVerifier) Verify(token string) (*Identity, error) {
 
 	metrics.Get().StsResponses.WithLabelValues(fmt.Sprint(response.StatusCode)).Inc()
 	if response.StatusCode != 200 {
-		return nil, NewSTSError(fmt.Sprintf("error from AWS (expected 200, got %d). Body: %s", response.StatusCode, string(responseBody[:])))
+		responseStr := string(responseBody[:])
+		if strings.Contains(responseStr, "Throttling") {
+			return nil, STSThrottling{}
```

Comment:
Should we need to add a message here for this?

Reply:
Yes, I think we need to pass through the error here.

Reply:
thank you! updated